os/drivers/Kconfig

```diff
@@ -154,6 +154,7 @@ config ARCH_HAVE_I2CRESET
 
 menuconfig I2C
 	bool "I2C Driver Support"
+	depends on ARCH_HAVE_I2C
```

Comment:
Is this better? Logicalliy this is needed. But this module does not have any dependancy with HW (arch). Other drivers like GPIO, SPI, ... does not have any dependancy with ARCH. If you want to do this, we had better adapt same policy.